.github/actions/docker-build-and-push-tools/action.yaml

```diff
@@ -0,0 +1,96 @@
+name: docker-build-and-push-tools
+description: Composite action to build and push tools images to registry.
+
+inputs:
+  platform:
+    description: Target platform.
+    required: true
+  target-image:
+    description: Target docker image name in the registry.
+    required: true
+  build-args:
+    description: Additional build args.
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Install jq and vcstool
+      run: |
+        sudo apt-get -y update
+        sudo apt-get -y install jq python3-pip
+        pip install --no-cache-dir vcstool
+      shell: bash
+
+    - name: Run vcs import
+      run: |
+        vcs import src < autoware.repos
+        vcs import src < simulator.repos
+      shell: bash
+
+    - name: Setup Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Restore ccache
+      uses: actions/cache/restore@v4
+      with:
+        path: |
+          root-ccache
+        key: ccache-${{ inputs.platform }}-${{ hashFiles('src/**/*.cpp') }}
+        restore-keys: |
+          ccache-${{ inputs.platform }}-
+
+    - name: Restore apt-get
+      uses: actions/cache/restore@v4
+      with:
+        path: |
+          var-cache-apt
+        key: apt-get-${{ inputs.platform }}-${{ hashFiles('src/**/package.xml') }}
+        restore-keys: |
+          apt-get-${{ inputs.platform }}-
+
+    - name: Inject cache into docker
+      uses: reproducible-containers/buildkit-cache-dance@v3.1.2
+      with:
+        cache-map: |
+          {
+            "root-ccache": "/root/.ccache",
+            "var-cache-apt": "/var/cache/apt"
+          }
+        skip-extraction: true
+
```

Comment:
The cache is meant for the `Dockerfile`, so no cache is effective for `Dockerfile.visualizer`.

```suggestion
```